setup.py

```diff
@@ -690,6 +690,8 @@ def build_extensions(self) -> None:
                     feature.set("zlib", "z")
                 elif sys.platform == "win32" and _find_library_file(self, "zlib"):
                     feature.set("zlib", "zlib")  # alternative name
+                elif sys.platform == "win32" and _find_library_file(self, "zdll"):
+                    feature.set("zlib", "zdll")  # different name if shared
```

Comment:
So then the different name is not because it is shared?

Reply:
As far as I understand it is, the `.dll` library is the shared library and `zdll.lib` helps linking to it, if zlib is built as static library I think only `zlib.lib` is generated on Windows.
The reason nobody tried this is that zlib as shared library is not a common use case, most people use static libraries on Windows, in particular for smaller ones.

Reply:
Good point. Perhaps
```suggestion
                    feature.set("zlib", "zdll")  # dll import library
```

Reply:
> As far as I understand it is, the `.dll` library is the shared library and `zdll.lib` helps linking to it, if zlib is built as static library I think only `zlib.lib` is generated on Windows.

Correct, see Microsoft Docs : https://learn.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-creation#creating-an-import-library